client.go

```diff
@@ -64,25 +66,33 @@ func buildAzureClient() (*AzureClient, error) {
 		return nil, fmt.Errorf("Error building ARM Client: %s", err)
 	}
 
+	api, err := environments.EnvironmentFromString("")
+	if err != nil {
+		return nil, fmt.Errorf("unable to find environment %q from endpoint %q: %+v", environmentName, endpoint, err)
+	}
+
 	sender := sender.BuildSender("Azure Dalek")
 
 	oauthConfig, err := client.BuildOAuthConfig(environment.ActiveDirectoryEndpoint)
 	if err != nil {
 		return nil, err
 	}
 
-	resourceManagerAuth, err := client.GetAuthorizationToken(sender, oauthConfig, environment.TokenAudience)
+	resourceManagerAuth, err := client.GetMSALToken(context.Background(), api.ResourceManager, sender, oauthConfig, environment.TokenAudience)
 	if err != nil {
 		return nil, err
 	}
 
-	resourcesClient := resources.NewGroupsClientWithBaseURI(environment.ResourceManagerEndpoint, client.SubscriptionID)
-	resourcesClient.Authorizer = resourceManagerAuth
+	resourcesClient := resourcegroups.NewResourceGroupsClientWithBaseURI(environment.ResourceManagerEndpoint)
+	resourcesClient.Client.Authorizer = resourceManagerAuth
+
+	locksClient := managementlocks.NewManagementLocksClientWithBaseURI(environment.ResourceManagerEndpoint)
+	locksClient.Client.Authorizer = resourceManagerAuth
 
-	locksClient := locks.NewManagementLocksClientWithBaseURI(environment.ResourceManagerEndpoint, client.SubscriptionID)
-	locksClient.Authorizer = resourceManagerAuth
+	managementClient := managementgroups.NewManagementGroupsClientWithBaseURI(environment.ResourceManagerEndpoint)
+	managementClient.Client.Authorizer = resourceManagerAuth
 
-	graphAuth, err := client.GetAuthorizationToken(sender, oauthConfig, environment.GraphEndpoint)
+	graphAuth, err := client.GetMSALToken(context.Background(), api.MsGraph, sender, oauthConfig, environment.TokenAudience)
```

Comment:
meaning this can become:

```suggestion
	graphAuth, err := client.GetMSALToken(ctx, api.MsGraph, sender, oauthConfig, environment.TokenAudience)
```